appGeo/src/app/player/player.component.ts

```diff
@@ -18,259 +18,284 @@ const FAILURE = -10
 })
 
 
+export class Player implements OnInit{
+    userID: number // An int
+    username: string // A String
+    //geolocation // Object that is retunred from the Nativescript plugin
+    location // A Coordinate Object
+    alive: number // A Boolean
+    votes: number // An int
+    chat_lists // List of Chat Objects that Player is a part of
 
-export class Player implements OnInit {
-  userID: number // An int
-  username: string // A String
-  location // A Coordinate Object will update according to what player dev does
-  alive: number // A Boolean int value
-  votes: number // An int
-  chat_lists: Array<Chat> // List of Chat Objects that Player is a part of
-
-  constructor(){}
-
-  init_Player(userID: number, username: string, location, alive: number){
-    //I pushed things here because did not want to deal with injection atm
-    this.userID = userID;
-    this.username = username;
-    this.location = location;//will change to what playerdev decides to do
-    this.alive = alive;
-    this.votes = 0;
-  }
-
-  ngOnInit(): void {
-    //calls you want to make some base initialization when data is loaded in
-  }
-
-  getUserID(){
-    return this.userID;
-  }
-  getUsername() : string {
-    return this.username;
-  }
-  getLocation(){
-    return this.location;
-  }
-
-  getAliveStatus(){
-    return this.alive;
-  }
-  getVotes(){
-    return this.votes;
-  }
-
-  getChatList(){
-    return this.chat_lists;
-  }
-
-  getChat(chatID){
-    for(var i = 0; i < this.chat_lists.length; i++){
-        var curr_chat = this.chat_lists[i]
-        if (curr_chat.getChatID() == chatID)
-            return this.chat_lists[i];
+    constructor(){
+      this.votes = 0;
+      this.chat_lists = new Array()
```

Comment:
Note: I had to change the constructor to be an empty parameter taker because otherwise there was a ton of error with Injection being required (because in the Game file we call Instanceof which raises concern of the parameter being used as a value rather than input)